### PR TITLE
Add/preview options component readme

### PR DESCRIPTION
**Pull request by JustinyAhin**
## Description
Added documentation for the preview options component (see #22891)

## How has this been tested?
N/A, documentation only

## Types of changes
Documentation

## Checklist:
- [n/a] My code is tested.
- [n/a] My code follows the WordPress code style. 
- [n/a] My code follows the accessibility standards. 
- [n/a] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/preview-options/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+2. [Related components](#related-components)
+
+## Development guidelines
+
+### Usage
+
+Renders the previews options of the editor in a dropdown menu.
+
+```jsx
+import { PreviewOptions } from '@wordpress/block-editor';
```
Comment (tellthemachines) [inline packages/block-editor/src/components/preview-options/README.md:21]: This component is still being exported as `__experimental`; we should probably document its usage as `import { __experimentalPreviewOptions as PreviewOptions } from '@wordpress/block-editor';`. There is some precedent for this in other experimental component readmes, though we'll have to then remember to update the docs once it's moved out of experimental status 😅

File: packages/block-editor/src/components/preview-options/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+const MyPreviewOptions = () => (
+    <PreviewOptions
+        isEnabled={ true }
+        className="edit-post-post-preview-dropdown"
+        deviceType={ deviceType }
+        setDeviceType={ setPreviewDeviceType }
+    >
+);
```
Comment (tellthemachines) [inline packages/block-editor/src/components/preview-options/README.md:30]: Given that this component accepts `children`, should we have an example documenting that? We have [one instance](https://github.com/WordPress/gutenberg/blob/3e57ffdb824b0a313cfbebf812db5ce1bb068d2a/packages/edit-post/src/components/device-preview/index.js#L33) of its use in that format so far.

File: packages/block-editor/src/components/preview-options/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+### Props
+
+#### className
+
+The CSS classes added to the component.
+
+-   Type: any
```
Comment (tellthemachines) [inline packages/block-editor/src/components/preview-options/README.md:39]: This should be Type: `String`.

File: packages/block-editor/src/components/preview-options/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+And example of when the preview options are not enabled is when the current post is not savable.
+
+-   Type: `boolean`
+-   Default: true
+
+#### deviceType
+
+The device type in the preview options. It can be either Desktop or Tablet or Mobile among others.
+
+-   Type: `any`
```
Comment (tellthemachines) [inline packages/block-editor/src/components/preview-options/README.md:53]: This should be Type: `String`.

File: packages/block-editor/src/components/preview-options/README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+#### setDeviceType
+
+Used to set the device type that will be used to display the preview inside the editor.
+
+-   Type: `any`
```
Comment (tellthemachines) [inline packages/block-editor/src/components/preview-options/README.md:59]: This should be Type:  `func`.

File: packages/block-editor/src/components/preview-options/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    <PreviewOptions
+        isEnabled={ true }
+        className="edit-post-post-preview-dropdown"
+        deviceType={ deviceType }
+        setDeviceType={ setPreviewDeviceType }
+    >
+);
+```
+
+### Props
```
Comment (tellthemachines) [inline packages/block-editor/src/components/preview-options/README.md:33]: Might be good to also add to each prop whether it is Required; in this case only `deviceType` and `setDeviceType` are required props; the rest are optional.

**Review by tellthemachines — CHANGES_REQUESTED** [review]
Thanks for working on this! I've left a few comments below.

File: packages/block-editor/src/components/preview-options/README.md (hunk 1 above)
Comment (JustinyAhin) [reply packages/block-editor/src/components/preview-options/README.md:21]: So, it's fine to document experimental components I guess.

File: packages/block-editor/src/components/preview-options/README.md (hunk 2 above)
Comment (JustinyAhin) [reply packages/block-editor/src/components/preview-options/README.md:30]: Hum 🤔, I guess we can do that. For the past docs PRs I sent, I tried to keep it short and only add examples for the component itself.

Comment (tellthemachines) [reply packages/block-editor/src/components/preview-options/README.md:30]: Are you happy to do this? Not a blocker, more of a nice-to-have 🙂

**Review by tellthemachines — APPROVED** [review]
Thanks for the quick iteration, looks good!

Not sure why the Compressed Size check is failing, doesn't seem related to this changeset 🤔 Might be worth rebasing if we can't get it to pass by re-running the checks.

File: packages/block-editor/src/components/preview-options/README.md (hunk 2 above)
Comment (JustinyAhin) [reply packages/block-editor/src/components/preview-options/README.md:30]: Sure thing.
I can do something like

```jsx
import { Icon, MenuGroup } from '@wordpress/components';
import { PostPreviewButton } from '@wordpress/editor';
import { __experimentalPreviewOptions as PreviewOptions } from '@wordpress/block-editor';

const MyPreviewOptions = () => (
    <PreviewOptions
        isEnabled={ true }
        className="edit-post-post-preview-dropdown"
        deviceType={ deviceType }
        setDeviceType={ setPreviewDeviceType }
        >
        <MenuGroup>
            <div className="edit-post-header-preview__grouping-external">
                <PostPreviewButton
                    className={
                        'edit-post-header-preview__button-external'
                    }
                    role="menuitem"
                    forceIsAutosaveable={ hasActiveMetaboxes }
                    forcePreviewLink={ isSaving ? null : undefined }
                    textContent={
                        <>
                            { __( 'Preview in new tab' ) }
                            <Icon icon={ external } />
                        </>
                    }
                />
            </div>
		</MenuGroup>
    >
);
```

using the `DevicePreview` example.
Looks good?

**Comment by JustinyAhin** [comment]
> Thanks for the quick iteration, looks good!
> 
> Not sure why the Compressed Size check is failing, doesn't seem related to this changeset thinking Might be worth rebasing if we can't get it to pass by re-running the checks.

Wiil see after the next commit if it'll pass.

File: packages/block-editor/src/components/preview-options/README.md (hunk 2 above)
Comment (tellthemachines) [reply packages/block-editor/src/components/preview-options/README.md:30]: Sure, that works! One thing: the snippet above is missing the closing `</PreviewOptions>` tag; should add it in for clarity.

**Pull request merged**